### PR TITLE
Some enhancements

### PR DESCRIPTION
**Pull request by tcoratger**
# Pull Request type

Time spent on this PR: 45 min

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

## What is the new behavior?

Some enhancements and simplifications globally.

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: crates/build-utils/src/content_reader.rs [hunk 1]
```diff
@@ -52,7 +51,7 @@ impl ContentReader {
                 let sender = transaction
                     .get("sender")
                     .and_then(|value| value.as_str())
-                    .ok_or_else(|| eyre::eyre!("Key 'sender' not found"))?;
+                    .ok_or_else(|| eyre!("Key 'sender' not found"))?;
```
Comment (greged93) [inline crates/build-utils/src/content_reader.rs:54]: does `eyre` have `ok_or_else_eyre`?

**Review by greged93 — COMMENTED** [review]
just one small nit

File: crates/build-utils/src/content_reader.rs (hunk 1 above)
Comment (tcoratger) [reply crates/build-utils/src/content_reader.rs:54]: don't think so unfortunately.

**Review by greged93 — APPROVED** [review]
lgtm

**Pull request merged**